explainers/signed-exchange-subresource-subtitution-explainer.md

```diff
@@ -0,0 +1,455 @@
+   (= inner request URL and inner response) to the SignedExchangeCache.
+1. While
+[navigating across documents](https://html.spec.whatwg.org/multipage/browsing-the-web.html#navigating-across-documents),
+copy the parsedExchanges in the SignedExchangeCache of the source document to
+the PrefetchedSignedExchangeCache of the target document. This is intended to
+provide a way to pass the cached signed exchange across origins even if the UA
+is using origin isolated HTTPCache mechanism.
+1. While processing
+[preload](https://html.spec.whatwg.org/multipage/links.html#link-type-preload)
+link HTTP headers (eg: Link: <https://cdn.publisher.example/lib.js>;
+rel="preload"; as="script"):
+   - Check whether matching "allowed-alt-sxg" link HTTP header
+   (`Link: <https://cdn.publisher.example/lib.js>;rel="allowed-alt-sxg";header-integrity="sha256-.."`)
+   exists or not. (Note that if the allowed-alt-sxg link HTTP header has
+   variants and variant-key attributes, the UA must execute the algorithm
+   written in
+   [HTTP Representation Variants spec](https://httpwg.org/http-extensions/draft-ietf-httpbis-variants.html)
+   to find the matching header.)
+   - Check whether all preload links which have matching allowed-alt-sxg link
+   header have matching (url and header-integrity) parsedExchange in
+   PrefetchedSignedExchangeCache. If the check passes, set request's stashed
+   exchange to the parsedExchange so the resource will be loaded from the cached
+   signed exchange.
+   - After processing the link HTTP headers, clears the
+   PrefetchedSignedExchangeCache.
+
+## Recursive subresource signed exchange prefetch
+
+While prefetching an HTML resource in signed exchange format:
+
+1. When the UA detects a "preload" link HTTP header in the inner response, check
+whether a matching “allowed-alt-sxg” link HTTP header in the inner response
+exists or not. (Note that multiple `allowed-alt-sxg` links can be present for
+the same preload if they include `variants` and `variant-key` attributes. In
+that case, the UA uses the algorithm written in
+[HTTP Representation Variants](https://httpwg.org/http-extensions/draft-ietf-httpbis-variants.html)
+spec to find the matching header.)
+1. If an `allowed-alt-sxg` link exists, check whether the signed exchange was
+served with a matching “alternate” link HTTP header.
+1. If the outer signed exchange did identify an alternate version of the
+subresource, prefetch the subresource signed exchange instead of prefetching the
+original resource URL.
+1. The prefetched signed exchange will be stored to the SignedExchangeCache of
+the Document. And it will be passed to the next Document and used while
+processing the preload link header. This behavior is written in above.
+
+
+# Detailed design discussion
+## Header integrity of signed exchange
+The `header-integrity` parameter of the `allowed-alt-sxg` link is the SHA256
```

Comment:
There should probably be a section that starts with a definition of the `allowed-alt-sxg` link relation, including an example header expressing it.

Reply:
Done.
How about this?